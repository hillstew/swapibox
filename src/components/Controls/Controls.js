import React from "react";
import PropTypes from "prop-types";

 const Controls = ({ handleClick }) => {
  return (
    <div className="controls-div">
      <button onClick={handleClick} name="vehicles">
        <span>VEHICLES</span>
      </button>
      <button onClick={handleClick} name="planets">
        <span>PLANETS</span>
      </button>
      <button onClick={handleClick} name="people">
        <span>PEOPLE</span>
      </button>
    </div>
  );
};

Controls.propTypes = {
  handleClick: PropTypes.func.isRequired
}

export default Controls;
