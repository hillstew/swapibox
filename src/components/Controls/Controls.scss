@font-face {
  font-family: "Galaxy";
  src: url(./fonts/SFdistantGalaxy.ttf);
}

.controls-div {
  margin-top: 200px;


  button {
    background: none;
    border: 4px solid #dad8d8f5;
    color: #eebe22a8;
    cursor: pointer;
    font-family: "Galaxy";
    font-size: 3rem;
    height: 300px;
    margin: 35px 15px;
    padding-left: 40px;
    transition-duration: 1s;
    width: 220px;

    &:hover {
      background-color: rgb(0, 0, 29);  
      cursor: pointer;
    }

    span {
      display: block;
      height: 50px;
      letter-spacing: 1.5px;
      transform: rotate(-90deg);
      width: 300px;
    }
  }
}
