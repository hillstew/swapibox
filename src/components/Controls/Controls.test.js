import React from "react";
import Controls from "../Controls/Controls"
import { shallow } from "enzyme"

const mockHandleChange = jest.fn;

describe("Controls", () => {
  it("should match snapshot", () => {
    const wrapper = shallow(
      <Controls handleChange={mockHandleChange} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
