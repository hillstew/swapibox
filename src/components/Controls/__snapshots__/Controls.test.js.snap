// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls should match snapshot 1`] = `
<div
  className="controls-div"
>
  <button
    name="vehicles"
    onClick={[Function]}
  >
    <span>
      VEHICLES
    </span>
  </button>
  <button
    name="planets"
    onClick={[Function]}
  >
    <span>
      PLANETS
    </span>
  </button>
  <button
    name="people"
    onClick={[Function]}
  >
    <span>
      PEOPLE
    </span>
  </button>
</div>
`;
